storage/country_tree.hpp

```diff
@@ -152,98 +88,28 @@ class CountryTree
 
   bool IsEmpty() const { return m_countryTree == nullptr; }
 
-  Node const & GetRoot() const
-  {
-    CHECK(m_countryTree, ());
-    return *m_countryTree;
-  }
+  Node const & GetRoot() const;
 
-  Node & GetRoot()
-  {
-    CHECK(m_countryTree, ());
-    return *m_countryTree;
-  }
+  Node & GetRoot();
 
-  Country & AddAtDepth(size_t level, Country const & value)
-  {
-    Node * added = nullptr;
-    if (level == 0)
-    {
-      ASSERT(IsEmpty(), ());
-      m_countryTree = std::make_unique<Node>(value, nullptr);  // Creating the root node.
-      added = m_countryTree.get();
-    }
-    else
-    {
-      added = m_countryTree->AddAtDepth(level, value);
-    }
-
-    ASSERT(added, ());
-    m_countryTreeHashTable.insert(make_pair(value.Name(), added));
-    return added->Value();
-  }
+  Country & AddAtDepth(size_t level, Country const & value);
 
   /// Deletes all children and makes tree empty
-  void Clear()
-  {
-    m_countryTree.reset();
-    m_countryTreeHashTable.clear();
-  }
+  void Clear();
 
   /// \brief Checks all nodes in tree to find an equal one. If there're several equal nodes
   /// returns the first found.
   /// \returns a poiter item in the tree if found and nullptr otherwise.
-  void Find(CountryId const & key, std::vector<Node const *> & found) const
-  {
-    found.clear();
-    if (IsEmpty())
-      return;
-
-    if (key == m_countryTree->Value().Name())
-      found.push_back(m_countryTree.get());
-
-    auto const range = m_countryTreeHashTable.equal_range(key);
-    if (range.first == range.second)
-      return;
+  void Find(CountryId const & key, std::vector<Node const *> & found) const;
 
-    std::for_each(range.first, range.second,
-                  [&found](typename CountryTreeHashTable::value_type const & node) {
-                    found.push_back(node.second);
-                  });
-  }
-
-  Node const * const FindFirst(CountryId const & key) const
-  {
-    if (IsEmpty())
-      return nullptr;
-
-    std::vector<Node const *> found;
-    Find(key, found);
-    if (found.empty())
-      return nullptr;
-    return found[0];
-  }
+  Node const * const FindFirst(CountryId const & key) const;
 
   /// \brief Find only leaves.
   /// \note It's a termprary fucntion for compatablity with old countries.txt.
   /// When new countries.txt with unique ids will be added FindLeaf will be removed
   /// and Find will be used intead.
   /// @TODO(bykoianko) Remove this method on countries.txt update.
-  Node const * const FindFirstLeaf(CountryId const & key) const
-  {
-    if (IsEmpty())
-      return nullptr;
-
-    std::vector<Node const *> found;
-    Find(key, found);
-
-    for (auto node : found)
-    {
-      if (node->ChildrenCount() == 0)
-        return node;
-    }
-    return nullptr;
-  }
+  Node const * const FindFirstLeaf(CountryId const & key) const;
 
 private:
   using CountryTreeHashTable = std::multimap<CountryId, Node *>;
```

Comment:
s/HashTable/Map/g

and please drop this alias

Reply:
done